internal/controller/terraform.go

```diff
@@ -0,0 +1,47 @@
+package controller
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func (clctrl *ClusterController) TerraformPrep() error {
+
+	path := clctrl.ProviderConfig.GitopsDir + "/terraform"
+	err := filepath.Walk(path, detokenizeterraform(path, clctrl))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func detokenizeterraform(path string, clctrl *ClusterController) filepath.WalkFunc {
```

Comment:
If you decide to merge these functions that are almost a copy of each other, would you be interested in working together to add some unit tests here? We can make it a bit more resilient that way.

Reply:
sure Patrick,we can sync after todays standup